src/checkers/flashchecker.py

```diff
@@ -0,0 +1,118 @@
+# Flash Checker: A checker that to see if the url is pointing to the latest Flash Player.
+#
+# Consult the README for information on how to use this checker.
+#
+# Copyright © 2019 Endless Mobile, Inc.
+#
+# Authors:
+#       Ryan Gonzalez <rymg19@gmail.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+
+import json
+import logging
+import urllib.error
+import urllib.request
+import urllib.parse
+
+from lib.externaldata import ExternalData, ExternalFile, Checker
+from lib import utils
+
+log = logging.getLogger(__name__)
+
+
+FLASH_BASE_URL = 'http://get.adobe.com/flashplayer/webservices/json/?{params}'
+
+FLATPAK_TO_FLASH_ARCH_MAP = {
+    'i386': 'x86-32',
+    'x86_64': 'x86-64',
+}
+
+BROWSER_TO_PAPI_MAP = {
+    'Chrome': 'ppapi',
+    'Firefox': 'npapi',
+}
+
+
+class FlashChecker(Checker):
+    def _should_check(self, external_data):
+        return external_data.checker_data.get('type') == 'flash'
+
+    def _get_arches(self, external_data):
+        if len(external_data.arches) != 1:
+            return None
+
+        flatpak_arch = external_data.arches[0]
+        return flatpak_arch, FLATPAK_TO_FLASH_ARCH_MAP.get(external_data.arches[0])
+
+    def check(self, external_data):
+        if not self._should_check(external_data):
+            log.debug('%s is not a flash type ext data', external_data.filename)
+            return
+
+        browser = external_data.checker_data['browser'].title()
+
+        try:
+            papi = BROWSER_TO_PAPI_MAP[browser]
+        except KeyError:
+            log.warning('%s has an invalid browser (should be Chrome or Firefox)',
+                        external_data.filename)
+            return
+
+        arches = self._get_arches(external_data)
+        if arches is None:
+            log.warning('%s has invalid only-arches (should be one of %s)',
+                        external_data.filename,
+                        ', '.join(f'[{arch}]' for arch in FLATPAK_TO_FLASH_ARCH_MAP))
```

Comment:
Why wrap these in '[]'?

Reply:
Since `only-arches` has to be a list, this is supposed to show the YAML-like syntax for having a one-item list of the desired values. If you want, I can change it to add double-quotes, like JSON syntax.